packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/add/add.controller.js

```diff
@@ -6,47 +8,184 @@ export default class {
+        !destinationEngines.includes(this.database.engine)
+          ? [this.database]
+          : this.servicesList.filter((db) => sourceEngines.includes(db.engine));
+      this.destinationServiceList =
+        isEqual(destinationEngines, [this.database.engine]) &&
+        !sourceEngines.includes(this.database.engine)
+          ? [this.database]
+          : this.servicesList.filter((db) =>
+              destinationEngines.includes(db.engine),
+            );
+    }
+    // Preselect service if only one element
+    this.sourceService =
+      this.sourceServiceList.length === 1 ? this.sourceServiceList[0] : null;
+    this.destinationService =
+      this.destinationServiceList.length === 1
+        ? this.destinationServiceList[0]
+        : null;
+    // Emtpy parameters array
+    this.parameters = {};
+    // Disable select inputs if needed
+    const isServiceSourceListEmpty = this.sourceServiceList.length === 0;
+    const isCurrentEngineInSourceList = sourceEngines.includes(
+      this.database.engine,
+    );
+    const isServiceDestinationListEmpty =
+      this.destinationServiceList.length === 0;
+    const isCurrentEngineInDestinationList = destinationEngines.includes(
+      this.database.engine,
+    );
+    // Disable select input if needed
+    this.isDisabledSourceSelect =
+      isServiceSourceListEmpty ||
+      (isCurrentEngineInSourceList && !isCurrentEngineInDestinationList);
+    this.isDisabledDestinationSelect =
+      isServiceDestinationListEmpty ||
+      (isCurrentEngineInDestinationList && !isCurrentEngineInSourceList);
+    // Check if preselected data is valid
+    this.checkIntegrationValidity();
+  }
+
+  checkIntegrationValidity() {
+    // We can not add twice the same integration
+    this.existingIntegrationError =
+      this.serviceIntegrationList.filter(
+        (existingIntegration) =>
+          existingIntegration.type === this.integrationCapability?.type &&
+          existingIntegration.sourceServiceId === this.sourceService?.id &&
+          existingIntegration.destinationServiceId ===
+            this.destinationService?.id,
+      ).length > 0;
+    // We have to select the current service as either source or destination
+    this.notCurrentServiceSelectedError =
+      this.sourceService?.id !== this.database.id &&
+      this.destinationService?.id !== this.database.id;
+    // Show a message if no service matches the selected integration type
+    this.showNoServiceMessage =
+      this.sourceServiceList.length === 0 ||
+      this.destinationServiceList.length === 0;
+    // List of expected engines to add for current integration type
+    this.expectedEngines =
+      this.sourceServiceList.length === 0
+        ? this.integrationCapability.sourceEngines
+        : this.integrationCapability.destinationEngines;
+  }
+
+  formatParameters() {
+    // Do not post parameter property at all if no parameter
+    if (Object.keys(this.parameters).length === 0) {
+      return undefined;
```

Comment:
```suggestion
      return;
```

Reply:
Linter does not allow this because of this rule: https://eslint.org/docs/latest/rules/consistent-return